nipoppy_cli/nipoppy/config/main.py

```diff
@@ -22,41 +23,63 @@ class Config(ModelWithContainerConfig):
+            "of the config file. Note: the replacement only happens if the config "
+            "is loaded from a file with :func:`nipoppy.config.main.Config.load`"
+        ),
     )
-    PROC_PIPELINES: dict[str, dict[str, PipelineConfig]] = Field(
+    BIDS_PIPELINES: list[BidsPipelineConfig] = Field(
+        default=[], description="Configurations for BIDS conversion, if applicable"
+    )
+    PROC_PIPELINES: list[PipelineConfig] = Field(
         description="Configurations for processing pipelines"
     )
 
     model_config = ConfigDict(extra="allow")
 
     def _check_no_duplicate_pipeline(self) -> Self:
         """Check that BIDS and PROC_PIPELINES do not have common pipelines."""
-        bids_pipelines = set(self.BIDS.keys())
-        proc_pipelines = set(self.PROC_PIPELINES.keys())
-        if len(bids_pipelines & proc_pipelines) != 0:
-            raise ValueError(
-                "Cannot have the same pipeline under BIDS and PROC_PIPELINES"
-                f", got {bids_pipelines} and {proc_pipelines}"
-            )
+
+        def _check_pipeline_infos(
+            pipeline_configs: list[PipelineConfig],
+            pipeline_type: str,
+            info_func: Callable[[PipelineConfig | BidsPipelineConfig], Tuple],
+        ):
+            pipeline_infos = set()
+            for pipeline_config in pipeline_configs:
+                pipeline_info = info_func(pipeline_config)
+                if pipeline_info in pipeline_infos:
+                    raise ValueError(
+                        f"Found multiple configurations for {pipeline_type} pipeline: "
+                        f"{pipeline_info}"
+                    )
+                pipeline_infos.add(pipeline_info)
+            return pipeline_infos
+
+        _check_pipeline_infos(
+            self.PROC_PIPELINES,
+            pipeline_type="processing",
+            info_func=lambda x: (x.NAME, x.VERSION),
+        )
+        _check_pipeline_infos(
+            self.BIDS_PIPELINES,
+            pipeline_type="BIDS conversion",
+            info_func=lambda x: (x.NAME, x.VERSION, x.STEP),
+        )
 
     def _propagate_container_config(self) -> Self:
         """Propagate the container config to all pipelines."""
 
-        def _propagate(pipeline_or_pipeline_dicts: dict | PipelineConfig):
-            if isinstance(pipeline_or_pipeline_dicts, PipelineConfig):
-                pipeline_config = pipeline_or_pipeline_dicts
+        def _propagate(pipeline_configs: list[PipelineConfig]):
+            for pipeline_config in pipeline_configs:
                 container_config = pipeline_config.get_container_config()
                 if container_config.INHERIT:
                     container_config.merge_args_and_env_vars(self.CONTAINER_CONFIG)
-            else:
-                for (
-                    child_pipeline_or_pipeline_dicts
-                ) in pipeline_or_pipeline_dicts.values():
-                    _propagate(child_pipeline_or_pipeline_dicts)
 
```

Comment:
Not sure how this works either.. currently there are no definitions found for `get_container_config()`